lib/jetstream/pull_consumer.ex

```diff
@@ -192,31 +298,64 @@ defmodule Jetstream.PullConsumer do
           message.gnat,
           settings.stream_name,
           settings.consumer_name,
-          settings.listening_topic
+          state.listening_topic
         )
 
       :noreply ->
         next_message(
           message.gnat,
           settings.stream_name,
           settings.consumer_name,
-          settings.listening_topic
+          state.listening_topic
         )
     end
 
     {:noreply, state}
   end
 
-  def handle_info(other, state) do
-    require Logger
+  def handle_info({:EXIT, _pid, _reason}, state) do
+    Logger.info(
+      """
+        #{__MODULE__} for #{state.settings.stream_name}.#{state.settings.consumer_name}: NATS connection has died.
+        PullConsumer is reconnecting.
+      """,
+      module: state.module,
+      listening_topic: state.listening_topic,
+      subscription_id: state.subscription_id,
+      connection_name: state.settings.connection_name
+    )
 
-    Logger.error("""
-    #{__MODULE__} for #{state.settings.stream_name}.#{state.settings.consumer_name} received unexpected message:
-    #{inspect(other)}
-    """)
+    {:connect, :reconnect, state}
+  end
+
+  def handle_info(other, state) do
+    Logger.error(
```

Comment:
```suggestion
    Logger.debug(
```

This is not an erroneous case and I think we should avoid log noise from this library.